community/kernel/src/main/java/org/neo4j/graphdb/notification/NotificationImpl.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphdb.notification;
+
+/**
+ * Implementation of {@link Notification}
+ */
+public class NotificationImpl implements Notification
+{
+    private final NotificationCode code;
+    private final String title;
+    private final String description;
+    private final NotificationSeverityLevel severity;
+    private final InputPosition position;
+
+    /**
+     * use {@link NotificationBuilder}
+     */
+    private NotificationImpl( NotificationCode code, String title, String description, NotificationSeverityLevel severity, InputPosition position )
+    {
+        this.code = code;
+        this.title = title;
+        this.description = description;
+        this.severity = severity;
+        this.position = position;
+    }
+
+    @Override
+    public NotificationCode getCode()
+    {
+        return code;
+    }
+
+    @Override
+    public String getTitle()
+    {
+        return title;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return description;
+    }
+
+    @Override
+    public NotificationSeverityLevel getSeverity()
+    {
+        return severity;
+    }
+
+    @Override
+    public InputPosition getPosition()
+    {
+        return position;
+    }
+
+    /**
+     * Builds a notification message.
+     */
+    public static class NotificationBuilder
```

Comment:
Since this is an inner class, I'd think it'd be sufficient to call it `Builder`, since then it would typically be referenced as `NotificationImpl.Builder`, which is clear enough (although naming a class "something Impl" should carry the death penalty).